src/services/expenseService.js

```diff
@@ -0,0 +1,52 @@
+let expenses = [];
+let expenseId = 1;
+
+const resetExpenses = () => {
+  expenses = [];
+};
+
+const getAllExpense = () => {
+  return expenses;
+};
+
+const getExpensesById = (id) => {
+  return expenses.find((expense) => expense.id === +id);
+};
+
+const createExpense = (userId, spentAt, title, amount, category, note) => {
+  const expense = {
+    id: expenseId,
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  };
+
+  expenseId++;
+  expenses.push(expense);
+
+  return expense;
+};
+
+const updateExpense = (id, data) => {
+  const expense = getExpensesById(id);
+
+  Object.assign(expense, data);
```

Comment:
The `updateExpense` function does not handle the case where the expense with the given ID is not found. Consider adding a check to return `null` or an appropriate response if the expense is not found before attempting to update it.